text/0070-metric-bound-instrument.md

```diff
@@ -0,0 +1,63 @@
+# Rename metric instrument Handles to "Bound Instruments"
+
+*Status: proposed 11/26/2019*
+
+The OpenTelemetry metrics API specification refers to a concept known
+as ["metric handles"](0009-metric-handles.md), which is a metric
+instrument bound to a `LabelSet`.  This OTEP proposes to change that
+term to "bound instruments" to avoid the more-generic term "handle".
+
+The corresponding method to create a bound instrument will be renamed
+"Bind" as opposed to "GetHandle".
+
+## Motivation
+
+The term "Handle" is widely seen as too general for its purpose in the
+metrics API.  Rather than re-use a widely-used noun for this concept,
+we instead will re-use the metric "instrument" noun and apply an
+adjective, "bound" to convey that it has been bound to a `LabelSet`.
+
+## Explanation
+
+"Handle" has been confusing from the start. However it was preceded by
+other potentially confusing terms (e.g., "TimeSeries", "Entry").  The
+term "Bound instrument" was initially suggested
+[here](https://github.com/open-telemetry/opentelemetry-specification/pull/299#discussion_r334211154)
+and widely accepted.
+
+## Internal details
+
+This is a simple renaming.  All uses of "handle" will be replaced by
+"bound instrument" in the specification.  All uses of the `GetHandle`
+method become `Bind`.
+
+Note that the phrase "bound instrument" may not appear directly in the
+user-facing API, nor is it required to, whereas the method `GetHandle`
+is a specified method on metric instruments.
+
+The newly-named `Bind()` method returns a bound instrument type.  The
+name of the returned type may simply take the name of its instrument
+with the prefix `Bound`.  For example, an `Int64Counter` instrument's
+`Bind()` method should return a `BoundInt64Counter` type.
```

Comment:
I don't really like the `BoundInt64Counter`. I would like to know what others think.

Reply:
Definitely a mouthful, but nothing better comes to mind 🤔

Reply:
I'm going to bet you would not prefer "Curried", which is another term used by Prometheus in this context.

Reply:
BoundInt64Instrument is a Go artifact. I expect it to show differently in other languages, eg BoundInstrument<long>, just BoundInstrument in Python/Js.

Reply:
I took @bogdandrutu's meaning to be that he does not like the "BoundX" construct entirely. I don't have any better suggestions.

Reply:
I am fine to keep it as is.